integration/eth2network/eth2_network.go

```diff
@@ -480,12 +482,15 @@ func (n *Impl) waitForMergeEvent(ctx context.Context, startTime time.Time) error
 		return err
 	}
 
-	// wait for the merge block (exit if context closes)
-	for ; number <= 7; time.Sleep(time.Second) {
+	// wait for the merge block
+	for timeoutStartTime := time.Now(); number <= 7; time.Sleep(time.Second) {
```

Comment:
If you're going to do it this way it'd be cleaner with `retry.Do` imo

Reply:
Done, not used to using the retry package, good shout.